src/Servers/IIS/AspNetCoreModuleV2/InProcessRequestHandler/StartupExceptionApplication.h

```diff
@@ -27,18 +29,20 @@ class StartupExceptionApplication : public InProcessApplicationBase
 
     HRESULT CreateHandler(IHttpContext *pHttpContext, IREQUEST_HANDLER ** pRequestHandler)
     {
-        *pRequestHandler = new ServerErrorHandler(*pHttpContext, 500, 30, "Internal Server Error", m_HR, m_moduleInstance, m_disableLogs, IN_PROCESS_RH_STATIC_HTML);
-        return S_OK;
-    }
+        if (m_errorPageContent.length() > 0)
```

Comment:
I know security is a concern here, so I propose we make this error page only surface when running behind IISExpress or the development environment variable is set.

Reply:
IIS Express is an interesting way of checking this. 

ASPNETCORE_DETAILEDERRORS is an existing variable we use for the hosting error page. I'd be leary about using Development.

Reply:
I did ASPNETCORE_ENVIRONMENT for now because it is the only this that is on by default as an environment variable. There are 3 different things we can key off of: DetailedErrors, CaptureStartupErrors, and Environment. I personally think CaptureStartupErrors is the best thing to use, but I don't see this as an environment variable anywhere.